pkg/dialer/dialer.go

```diff
@@ -29,51 +30,85 @@ type dialResult struct {
 	err error
 }
 
-// ContextDialer returns a GRPC net.Conn connected to the provided address
+var (
+	// ErrTimeout represents a timeout error while dialing
+	ErrTimeout = stderrs.New("timeout")
+)
+
+// ContextDialer returns a GRPC net.Conn connected to the provided address.
+// It tolerates ENOENT and keeps retrying if provided context has a deadline.
```

Comment:
> I don't understand why we would not tolerate ENOENT in the case where the context passed in does not have a deadline.

I explicitly add a hint saying "... if provided context has a deadline." because the original code does so, but the author didn't give any notification or warning about it. 
Actually, To avoid any unexpected failure or inconsistent behavior after this patch, just rewrite this function to make it "more context-aware" without introducing any observable change from caller's view.

Reply:
ok... not sure if the two additional lines of descriptive text for the function help or not